battery_safety_tester.py

```diff
@@ -0,0 +1,33 @@
+from math import isnan
+from Constant import SUPPORTED_SAFETY_LIMIT_CHECKS
+from reporter import report_parameters
+from sensor_value_handler import snap_limits_to_sensor_accuracy
+
+
+def check_battery_param_limits(battery_param_name, battery_param_value):
+    if battery_param_value <= SUPPORTED_SAFETY_LIMIT_CHECKS[battery_param_name]['limits']['min']:
+        return 'ALERT_UNDERLIMIT'
+    elif battery_param_value >= SUPPORTED_SAFETY_LIMIT_CHECKS[battery_param_name]['limits']['max']:
+        return 'ALERT_OVERLIMIT'
+    return 'OK'
+
+
+def check_battery_param_safety(battery_param_name, battery_param_value):
+    if isnan(battery_param_value):
+        return 'ALERT_VALUENAN'
+    elif battery_param_name not in SUPPORTED_SAFETY_LIMIT_CHECKS:
+        return 'ALERT_PARAMNAMEUNKNOWN'
+    return check_battery_param_limits(battery_param_name, battery_param_value)
+
+
+def check_overall_battery_safety(**all_battery_params):
```

Comment:
Consistency: Sometimes, check_returns the status as a string, sometimes a bool.
Also, does `check_overall_battery_safety` return `True` when the battery needs attention? Or when everything is ok?
These questions require us to go through the code to understand it. Can this be evident from the function name?

Reply:
Regarding the consistency, I have designed the overall check to return bool and individual checks to return string since I need granularity. Is that bad practice? If so, what would be acceptable in such scenarios? All string?

I agree that it is ambiguous and needs some renaming to better explain the return param. I will incorporate this. Thanks for pointing it out.